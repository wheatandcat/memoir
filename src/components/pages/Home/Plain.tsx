import React, { memo } from 'react';
import {
  ItemsByDateQuery as Query,
  ItemQueryVariables as Variables,
} from 'queries/api/index';
import ErrorPage from 'components/organisms/Error/Error';
import TemplateHome from 'components/templates/Home/Page';
import { QueryResult } from '@apollo/client';
import { ConnectedType } from './Connected';

type QueryHookResult = QueryResult<Query, Variables>;
type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;

type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;

  return <TemplateHome {...props} />;
};

export default memo<Props>(Plain);
