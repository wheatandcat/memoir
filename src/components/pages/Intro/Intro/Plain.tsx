import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import Loading from 'components/atoms/Loading';
import TemplateSettingMemoir from 'components/templates/Intro/Intro/Page';
import { State as MemoirNotificationSettingState } from 'hooks/useMemoirNotificationSetting';
import View from 'components/atoms/View';
import { ConnectedType } from './Connected';

type Props = ConnectedType & {
  data: MemoirNotificationSettingState;
  loading: boolean;
};

const Plain: React.FC<Props> = (props) => {
  if (props.loading)
    return (
      <View style={styles.root}>
        <Loading />
      </View>
    );

  return (
    <TemplateSettingMemoir
      onSaveNotification={props.onSaveNotification}
      dayOfWeek={props.data.dayOfWeek}
      hours={props.data.hours}
      minutes={props.data.minutes}
      notification={props.data.notification}
      onFinish={props.onFinish}
    />
  );
};

const styles = StyleSheet.create({
  root: {
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default memo(Plain);
