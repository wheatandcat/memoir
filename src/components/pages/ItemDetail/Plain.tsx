import React, { memo } from 'react';
import {
  ItemQuery as Query,
  ItemQueryVariables as Variables,
} from 'queries/api/index';
import ErrorPage from 'components/organisms/Error/Error';
import TemplateItemDetail from 'components/templates/ItemDetail/Page';
import { QueryResult } from '@apollo/client';
import { ConnectedType } from './Connected';

type QueryHookResult = QueryResult<Query, Variables>;
type QueryProps = Pick<QueryHookResult, 'data' | 'loading' | 'error'>;

export type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;

  const item = props.data?.item;

  return (
    <TemplateItemDetail
      {...props}
      title={item?.title || ''}
      categoryID={item?.categoryID || 0}
      like={item?.like || false}
      dislike={item?.dislike || false}
      itemDate={item?.date || ''}
    />
  );
};

export default memo<Props>(Plain);
