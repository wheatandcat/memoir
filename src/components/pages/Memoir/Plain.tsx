import React, { memo } from 'react';
import {
  ItemsInPeriodQuery as Query,
  ItemsInPeriodQueryVariables as Variables,
} from 'queries/api/index';
import ErrorPage from 'components/organisms/Error/Error';
import { ConnectedType } from './Connected';
import TemplateMemoir from 'components/templates/Memoir/Page';
import { Item, ItemsInPeriodPageInfo } from 'hooks/useItemsInPeriodPaging';
import { QueryResult } from '@apollo/client';

type QueryHookResult = QueryResult<Query, Variables>;
type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;

export type Props = QueryProps &
  ConnectedType & {
    items: Item[];
    pageInfo: ItemsInPeriodPageInfo;
  };

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;
  if (props.loading && props.items.length === 0 && !props.isFilter) {
    return null;
  }

  return (
    <TemplateMemoir
      startDate={props.startDate}
      endDate={props.endDate}
      loading={props.loading}
      search={props.search}
      items={props.items}
      users={props.users}
      selectedUserIDList={props.selectedUserIDList}
      onLoadMore={props.onLoadMore}
      pageInfo={props.pageInfo}
      onScreenShot={props.onScreenShot}
      onChangeUserID={props.onChangeUserID}
    />
  );
};

export default memo<Props>(Plain);
