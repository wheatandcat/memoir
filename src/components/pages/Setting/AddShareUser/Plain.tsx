import React, { memo } from 'react';
import ErrorPage from 'components/organisms/Error/Error';
import Loading from 'components/atoms/Loading';
import TemplateSettingAddShareUser from 'src/components/templates/Setting/AddShareUser/Page';
import { ConnectedType } from './Connected';

type QueryHookResult = any;
export type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;
export type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;
  if (props.loading) return <Loading />;

  return <TemplateSettingAddShareUser />;
};

export default memo(Plain);
