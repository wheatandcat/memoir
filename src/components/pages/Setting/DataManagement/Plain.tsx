import React, { memo } from 'react';
import ErrorPage from 'components/organisms/Error/Error';
import TemplateDataManagement from 'components/templates/Setting/DataManagement/Page';
import { ConnectedType } from './Connected';

type QueryHookResult = any;
type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;
export type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;

  return (
    <TemplateDataManagement
      loading={props.loading}
      onDelete={props.onDelete}
      disabledDeleteButton={props.disabledDeleteButton}
    />
  );
};

export default memo(Plain);
