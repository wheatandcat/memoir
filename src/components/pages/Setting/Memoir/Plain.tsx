import React, { memo } from 'react';
import ErrorPage from 'components/organisms/Error/Error';
import Loading from 'components/atoms/Loading';
import TemplateSettingMemoir from 'components/templates/Setting/Memoir/Page';
import { ConnectedType } from './Connected';

type QueryHookResult = any;
export type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;
export type Props = QueryProps & ConnectedType;

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;
  if (props.loading) return <Loading />;

  return <TemplateSettingMemoir onSave={props.onSave} />;
};

export default memo(Plain);
