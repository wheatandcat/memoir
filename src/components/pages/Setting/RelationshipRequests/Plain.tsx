import React, { memo } from 'react';
import ErrorPage from 'components/organisms/Error/Error';
import { RelationshipRequestsQueryHookResult as QueryHookResult } from 'queries/api/index';
import Loading from 'components/atoms/Loading';
import {
  RelationshipRequest,
  RelationshipRequestsPageInfo,
} from 'hooks/useRelationshipRequestsPaging';
import TemplateSettingRelationshipRequests from 'components/templates/Setting/RelationshipRequests/Page';
import { ConnectedType } from './Connected';

export type QueryProps = Pick<QueryHookResult, 'loading' | 'error'>;
export type Props = QueryProps &
  ConnectedType & {
    items: RelationshipRequest[];
    pageInfo: RelationshipRequestsPageInfo;
  };

const Plain: React.FC<Props> = (props) => {
  if (props.error) return <ErrorPage error={props.error} />;
  if (props.loading && props.items.length === 0) {
    return <Loading />;
  }

  return (
    <TemplateSettingRelationshipRequests
      items={props.items}
      onLoadMore={props.onLoadMore}
      pageInfo={props.pageInfo}
      loading={props.loading}
      onOK={props.onOK}
      onNG={props.onNG}
    />
  );
};

export default memo(Plain);
