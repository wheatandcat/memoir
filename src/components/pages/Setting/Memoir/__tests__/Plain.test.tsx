import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import { memoirNotificationSetting } from '__mockData__/memoirNotificationSetting';
import Plain, { Props } from '../Plain';

const propsData = (): Props => ({
  loading: false,
  onSave: jest.fn(),
  data: memoirNotificationSetting(),
});

describe('components/pages/Setting/Memoir/Plain.tsx', () => {
  beforeEach(() => {
    testRenderer(<Plain {...propsData()} />)();
  });

  it('正常にrenderすること', () => {
    expect(screen.findAllByText('')).toBeTruthy();
  });
});
