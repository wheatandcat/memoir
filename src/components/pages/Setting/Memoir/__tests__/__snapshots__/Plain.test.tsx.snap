// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Setting/Memoir/Plain.tsx 正常にrenderすること 1`] = `
<Memo(Page)
  dayOfWeek={1}
  hours={0}
  minutes={0}
  notification={false}
  onSave={[MockFunction]}
/>
`;
